scripts/dotnet-cli-build/PublishTargets.cs

```diff
@@ -436,5 +437,116 @@ public static BuildTargetResult TriggerDockerHubBuilds(BuildTargetContext c)
             }
             return c.Success();
         }
+
+        private const string PackagePushedSemaphoreFileName = "packages.pushed";
+
+        [Target(nameof(PrepareTargets.Init), nameof(InitPublish))]
+        public static BuildTargetResult PullNupkgFilesFromBlob(BuildTargetContext c)
+        {
+            Directory.CreateDirectory(Dirs.PackagesNoRID);
+
+            var hostBlob = $"{Channel}/Binaries/";
+
+            string forcePushBuild = Environment.GetEnvironmentVariable("FORCE_PUBLISH_BLOB_BUILD_VERSION");
+
+            if (!string.IsNullOrEmpty(forcePushBuild))
+            {
+                Console.WriteLine($"Forcing all nupkg packages for build version {forcePushBuild}.");
+                DownloadPackagesForPush(hostBlob + forcePushBuild);
+                return c.Success();
+            }
+
+            List<string> buildVersions = new List<string>();
+
+            Regex buildVersionRegex = new Regex(@"Binaries/(?<version>\d+\.\d+\.\d+-[^-]+-\d{6})/$");
+
+            foreach (string file in AzurePublisherTool.ListBlobs(hostBlob))
+            {
+                var match = buildVersionRegex.Match(file);
+                if (match.Success)
+                {
+                    buildVersions.Add(match.Groups["version"].Value);
+                }
+            }
+
+            // Sort decending
+            buildVersions.Sort();
+            buildVersions.Reverse();
+
+            // Try to publish the last 10 builds
+            foreach (var bv in buildVersions.Take(10))
+            {
+                Console.WriteLine($"Checking drop version: {bv}");
+
+                if (ShouldDownloadAndPush(hostBlob, bv))
+                {
+                    DownloadPackagesForPush(hostBlob + bv);
+                }
+            }
+
+            return c.Success();
+        }
+
+        private static bool ShouldDownloadAndPush(string hostBlob, string buildVersion)
+        {
+            // Set of runtime ids to look for to act as the signal that the build
+            // as finished each of these legs of the build.
+            Dictionary<string, bool> runtimes = new Dictionary<string, bool>()
```

Comment:
This is the list of rids that we run builds for? Could we move this to a more central location?